lib/orogen/spec/output_port.rb

```diff
@@ -135,6 +136,23 @@ def triggered_on_update?
             def triggered_once_per_update?
                 !!@triggered_once_per_update
             end
+
+            def init_policy?
+                @init_policy == true
+            end
+
+            # Calls keep_last_written_value(value) if value is a Boolean
+            def init_policy(value = nil)
+                return @init_policy if value.nil?
+
+                unless [true, false].include?(value)
+                    raise ArgumentError,
+                          "init_policy can only be called with " \
+                          "true or false. Got #{value.inspect}"
+                end
+                @init_policy = value
+                keep_last_written_value(value)
+            end
```

Comment:
Right.

Generally speaking in Ruby you don't validate boolean values against true/false. You just accept whatever is given to you and interpret it as a true or false value.

To check whether the method was given arguments, use a `*splat` argument which allows you to find that out

```
def init_policy?
   @init_policy
end

def init_policy(*value)
    return @init_policy if value.empty?
    raise ArgumentError, "some message" if value.size > 1

    @init_policy = value.first
    keep_last_written_value(value.first)
    self
end
```

Return `self` at the end on the setter case so that we can chain method calls, e.g.

```
.init_policy(true)
.needs_reliable_connection
```